test/com/esotericsoftware/kryo/serializers/GenericsTest.java

```diff
@@ -361,4 +374,25 @@ public String toString () {
 
 	}
 
+	static class ClassWithConflictingTypeArguments {
+		static final class A {
+			C<String> c;
+
+			public A (C<String> c) {
+				this.c = c;
+			}
+		}
+
+		static class B<R, V> implements C<V> {
+			R v;
```

Comment:
To prevent confusion, this `R` could be renamed from `v` to `r`.

```suggestion
			R r;
```

Reply:
Good catch. Renamed.